kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/clues.kt

```diff
@@ -25,11 +23,8 @@ inline fun <R> withClue(crossinline clue: () -> Any?, thunk: () -> R): R {
    try {
       collector.pushClue { clue.invoke().toString() }
       return thunk()
-      // this is a special control exception used by coroutines
-   } catch (t: TimeoutCancellationException) {
-      throw Exceptions.createAssertionError(clueContextAsString() + (t.message ?: ""), t)
    } catch (e: Exception) {
-      throw Exception(clueContextAsString() + (e.message ?: ""), e)
+      throw Exceptions.createAssertionError(clueContextAsString() + (e.message ?: ""), e)
```

Comment:
I don't think AssertionError is appropriate here - no Assertion has explicitly failed here, it's a different situation. I'd suggest creating another Exception specifically for this purpose, not reusing AssertionError.

Reply:
Sure, I'll work on this later